packages/plugins/c-sharp/c-sharp/src/visitor.ts

```diff
@@ -72,10 +223,22 @@ export class CSharpResolversVisitor extends BaseVisitor<CSharpResolversPluginRaw
     if (this._parsedConfig.emitJsonAttributes) {
       this.jsonAttributesConfiguration = getJsonAttributeSourceConfiguration(this._parsedConfig.jsonAttributesSource);
     }
+
+    this.compositionTypeToImplementationsMap = compositionTypesData.compositionTypeToImplementationsMap;
+    this.compositionTypeNames = compositionTypesData.compositionTypeNames;
   }
 
   public getImports(): string {
-    const allImports = ['System', 'System.Collections.Generic', 'System.ComponentModel.DataAnnotations'];
+    const allImports = [
+      'System',
+      'System.Collections.Generic',
+      'System.ComponentModel.DataAnnotations',
+      'System.Collections.Concurrent',
+      'System.Collections',
+      'System.Linq',
+      'System.Linq.Expressions',
+      'Newtonsoft.Json.Linq',
```

Comment:
Looking back at this, and I realised that having `Newtonsoft.Json` installed is a pre-requisite for getting this implementation to work. We are using `JToken.ToObject` as an intermediate step for deserializing CompositionType object in their correct concrete implementation. Do you have any suggestion on how this could be changed?

Reply:
@cicciodm 
We need to avoid such dependency. Current implementation forces users to use `Newtonsoft.Json` and it's not a good idea (in case if a user doesn't want to use any JSON stuff or use STJ).

So I see 2 options here:
1. Make converters logic as an additional configuration option for C# plugin settings to enable/disable such converters generation (in the case when `Newtonsoft.Json` is used);
2. Move converters logic to a separate NuGet package and then create a configuration option to this plugin to use your NuGet while generating (also Newtonsoft-specific option).

Personally, I think the second approach is better because it removes copy-pasted stuff from type to type and uses 1 centralized NuGet. But it also creates some responsibility to you for supporting such a library.

Reply:
@seclerp I have done some research, and I believe that there is a possibility to re-implement the custom parser only by using `System.Text.Json`, as the library provides an intermediate representation of untyped deserialized object, a generic `Deserialize<T>` method that could be used for deserialization, and custom Deserializer for composition type.

In theory, the dependency on `Newtonsoft.Json` can be removed.

In practice, I would not be able to do the work and testing necessary to get this to work before the holidays, or next January even.

I think that a viable path forward is to perform a check on the `Newtonsoft.Json` dependency and use the current implementation only on a successful check: this could unblock a number of devs in the short term. I can then add another issue for removing this dependency, and get to it in the future. Thoughts?

Reply:
@cicciodm in my previous message I was not clear with what I mean by "dependency" in our context. `Newtonsoft.Json` is not a problem in general, because other plugin code already using it for other purposes (basic serialization attributes, for example). The problem is in your usage of Newtonsoft: it is used for any type of configuration, even if a user chose `System.Text.Json` attributes, it will still generate Newtonsoft-related code and it will fall to compile on the user machine (because Newtonsoft NuGet should be installed in such case)

I agree with your proposal about temporal enabling your feature just for the `Newtonsoft.Json` configuration. But could you introduce configuration options for that? See attributes configuration for examples

Reply:
I understand what you mean, rather than saying "The dependency can be removed" I should have phrased it as "I can build an alternative based on `System.Text.Json`" 👍 

Adding a configuration option such as `emitCompositionTypes` and adding documentation about having to install the `Newtonsoft.Json` NuGet makes sense to me!